src/entity/user/User.ts

```diff
@@ -239,15 +243,23 @@ export type AddUserData = Pick<
   | 'acceptedMarketing'
   | 'timezone'
   | 'experienceLevel'
+  | 'language'
 >;
 export type AddUserDataPost = { referral: string } & AddUserData;
 export type UpdateUserEmailData = Pick<User, 'id' | 'email'>;
 type AddNewUserResult =
   | { status: 'ok'; userId: string }
   | { status: 'failed'; reason: UserFailErrorKeys; error?: Error };
 
+const checkLanguage = (language?: string): boolean => {
+  return !!language && validLanguages.includes(language as ContentLanguage);
+};
+
 const checkRequiredFields = (data: AddUserData): boolean => {
-  if (data?.username && !data?.experienceLevel) {
+  if (
+    data?.username &&
+    (!data?.experienceLevel || !checkLanguage(data?.language))
```

Comment:
Was just wondering if we should maybe split them into 2 internal ifs?

Reply:
Split it now